src/Mib/Iface.php

```diff
@@ -266,7 +268,9 @@ public function getSpeeds(Snmp $snmp) : iterable
      */
     public function getHcSpeeds(Snmp $snmp) : iterable
     {
-        return $snmp->walk(self::OID_HC_SPEED);
+        foreach ($snmp->walk(self::OID_HC_SPEED) as $key => $value) {
+            yield $key => $value * self::HC_SPEED_MULTIPLIER;
```

Comment:
I think that definitely yeah. To be honest I would like to remove the `getSpeeds` method and rename this to `getSpeeds` :D I think that we are just kinda fixing the SNMP in those case ... This lib should imo make it easier to fetch some data using SNMP and not just blindly wrap SNMP clients...You can always fetch OIDs directly if you want to (without using the MIB classes)

Reply:
I just think that it will be confusing to lib consumers. They should fix the case themselves if they need to (also there may be cases when they don't want to "fix" this).

There is a lot of things to be fixed in SNMP :D but IMO that's not responsibility of this library, it's not smart wrapper and our SNMP knowledge is not that deep to be able to fix it properly I dare to say 🤔

Reply:
Why wouldn't you want to "fix" this? The number is useless otherwise - it's some artificial number which means nothing in the real world, basically.

I think it should be ... why would we fix those things on every single place where we use the library when we can fix it in the library?

Reply:
If you're querying HC speeds that are defined as `An estimate of the interface's current bandwidth in units of 1,000,000 bits per second` you should return just that and nothing else. And same applies everywhere else.